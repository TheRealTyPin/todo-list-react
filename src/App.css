.App {
  font-family: "Work Sans", sans-serif;
  font-weight: 300;
  font-size: 1.5rem;
  color: var(--clr-light);
  display: grid;
  grid:
    "header header header header" auto
    "...... lists  active ......" auto
    "...... footer footer ......" 1fr /
    1fr minmax(100px, 300px) minmax(250px, 500px) 1fr;
  min-height: 100vh;
}

.App-header {
  grid-area: header;
  text-align: center;
  font-size: calc(7vw + 2rem);
  font-weight: 900;
  color: rgba(0, 0, 0, 0.1);
  letter-spacing: 2px;
  margin: -0.3em 0 0.5em;
}

.App > footer {
  grid-area: footer;
  margin: .5rem;
  margin-top: 5rem;
  text-align: center;
  font-size: 1rem;
  justify-self: center;
  align-self: end;
}
