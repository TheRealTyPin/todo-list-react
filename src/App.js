import React from 'react'
import './App.css'

import ListsOverview from './components/ListsOverview'
import TodoList from './components/TodoList'

function App() {
	return (
		<div className="App">
			<h1 className="App-header">Stuff I need to do</h1>
			<ListsOverview />
			<TodoList />
			<footer>
				This project was inspired by Kevin Powell&apos;s <a href="https://github.com/kevin-powell/todo-list-collab">todo-list-collab</a>
			</footer>
		</div>
	)
}

export default App
