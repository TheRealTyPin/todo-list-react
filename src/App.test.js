import React from 'react'
import { shallow } from 'enzyme'

import App from './App'

describe('App', () => {
	it('snapshot matches', () => {
		const wrapper = shallow(<App />)
		expect(wrapper).toMatchSnapshot()
	})

	it('renders header', () => {
		const wrapper = shallow(<App />)
		expect(wrapper.find('.App-header')).toHaveText('Stuff I need to do')
	})

	it('renders ListsOverview', () => {
		const wrapper = shallow(<App />)
		expect(wrapper.find('ListsOverview')).toExist()
	})

	it('renders TodoList', () => {
		const wrapper = shallow(<App />)
		expect(wrapper.find('TodoList')).toExist()
	})

	it('renders footer', () => {
		const wrapper = shallow(<App />)
		expect(wrapper.find('footer')).toContainMatchingElement('a[href="https://github.com/kevin-powell/todo-list-collab"]')
	})
})
