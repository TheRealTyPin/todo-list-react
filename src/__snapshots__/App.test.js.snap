// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot matches 1`] = `
<div
  className="App"
>
  <h1
    className="App-header"
  >
    Stuff I need to do
  </h1>
  <ListsOverview />
  <TodoList />
  <footer>
    This project was inspired by Kevin Powell's 
    <a
      href="https://github.com/kevin-powell/todo-list-collab"
    >
      todo-list-collab
    </a>
  </footer>
</div>
`;
